app/models/piece.rb

```diff
@@ -30,20 +41,40 @@ class Piece < ApplicationRecord
 
 
   def vertical_obstruction?(destination_y)
-    range = self.y_position < destination_y ? (self.y_position..destination_y) : (destination_y..self.y_position)
+    range = self.y_position < destination_y ? (self.y_position...destination_y) : (destination_y...self.y_position)
     game.pieces.where(
-        color: self.color,
-        x_position: self.x_position, 
+        # color: self.color,
+        x_position: self.x_position,
         y_position: range
       ).where.not(id: self.id).any?
   end
-  
+
   def horizontal_obstruction?(destination_x)
-    range = self.x_position < destination_x ? (self.x_position..destination_x) : (destination_x..self.x_position)
+    range = self.x_position < destination_x ? (self.x_position...destination_x) : (destination_x...self.x_position)
```

Comment:
(self.x_position...destination_x) - here you excluded the destination square however we need to check it also
(destination_x...self.x_position) - this works but might not necessary here because of this line 
.where.not(id: self.id).any? - checking that it's not a piece itself

Reply:
The reason I excluded destination square because the trello card is asking us to check between point A and B only, not the starting point or the destination point. I am thinking the reason is that some pieces can capture the piece at its destination square. And since I excluded the destination square, I changed the outcome of some of the tests from true to false as well.

Reply:
I see, that what I did in the beginning too. I'm just not sure how it will work if for example if white piece will try to move to square 4 from 2 and we'll have another white piece on sq 4 but our obstruction check will show that it's not obstructed....?

Reply:
This is where I actually got the ideas from: https://gist.github.com/kenmazaika/92b81db9e977578c8d94

If you look at the example A8 -> A6, it will show that is not an obstruction because there is nothing between the starting square and destination square. A piece can move before it reaches the destination square because there is nothing blocking it at all. It's easier to understand when you think of the obstructing piece as a piece sitting next to the piece we want to move, and that obstructing piece is preventing our piece from moving.

Reply:
Got it, thanks!